### PR TITLE
Changed the MCC to 4.11.0, Mbedtls to 2.27 and Parsec-se-driver to 0.6.0

### PR DESCRIPTION
**Pull request by prakharbhatt007**
# Mbed Edge pull request

Changes in this PR
* Mbed Cloud Client version 4.11.0
* Mbed TLS version 2.27.0
* Some Test file changes

### CODE REVIEW

File: Makefile.test [hunk 1]
```diff
@@ -35,7 +35,7 @@ run-tests-fota: build-test-fota-byoc
 	find ./build-test/bin -name "*-test" | xargs -n1 -I{} sh -c '{} || exit 255'
 
 run-tests-with-valgrind: build-test-byoc
-	find ./build-test/bin -name "*-test" | xargs -n1 -I{} sh -c 'valgrind --track-origins=yes --num-callers=100 --leak-check=full --show-reachable=yes --suppressions=./valgrind-unit-tests.supp --error-exitcode=255 {} || exit 255'
+	find ./build-test/bin -name "*-test" | xargs -n1 -I{} sh -c 'valgrind --track-origins=yes --num-callers=100 --leak-check=full --show-reachable=yes --error-exitcode=255 {} || exit 255'
```
Comment (ashishwigwag) [inline Makefile.test:38]: Wouldn't removing suppression generate issues?

**Review by ashishwigwag — APPROVED** [review]

File: lib/pal-platform/pal-platform.json [hunk 2]
```diff
@@ -210,11 +210,11 @@
     },
     "middleware": {
       "mbedtls": {
-        "version": "2.25.0",
+        "version": "2.27.0",
```
Comment (ygoyal18) [inline lib/pal-platform/pal-platform.json:213]: @prakharbhatt007 This needs to be changed for all targets not just RPi.

Comment (ygoyal18) [reply lib/pal-platform/pal-platform.json:213]: I have made that change.

File: lib/pal-platform/pal-platform.json [hunk 3]
```diff
@@ -228,11 +228,11 @@
         "to": "Middleware/trusted_storage/trusted_storage"
       },
       "parsec_se_driver": {
-        "version": "0.5.0",
+        "version": "0.6.0",
         "from": {
           "protocol": "git",
           "location": "https://github.com/parallaxsecond/parsec-se-driver.git",
-          "tag": "0.5.0"
+          "tag": "0.6.0"
```
Comment (JanneKiiskila) [inline lib/pal-platform/pal-platform.json:235]: We probably need to move upto 0.8.1, actually. @petedyerarm  @jenia81 @moshe-shahar

Comment (petedyerarm) [reply lib/pal-platform/pal-platform.json:235]: 0.6.0 is the latest version of the SE driver. 
0.8.1 is the version of the parsec-service.
So this is correct.

File: cmake/edge_configure.cmake [hunk 4]
```diff
@@ -152,6 +152,7 @@ if (${FIRMWARE_UPDATE})
   if (NOT FOTA_ENABLE)
      MESSAGE("Update client hub selected.")
      add_definitions ("-DMBED_CLOUD_CLIENT_SUPPORT_UPDATE=1")
+     add_definitions("-DPAL_DNS_API_VERSION=1")
```
Comment (JanneKiiskila) [inline cmake/edge_configure.cmake:155]: V2 gives the better IPv6 support, right? We should take that into use.

Comment (kivaisan) [reply cmake/edge_configure.cmake:155]: Why not to use the v3 which is already the default in 4.11 ?

Comment (prakharbhatt007) [reply cmake/edge_configure.cmake:155]: Getting this error. We tried on version 2 first but it's not supported on Linux. So V2 or greater is only supported on Mbed.
 #error "Async PAL DNS API v2 or greater is only supported on Mbed."

Comment (kivaisan) [reply cmake/edge_configure.cmake:155]: That's weird. Please investigate as according to 4.11 release notes:
`[Linux] Enabled PAL_DNS_API_VERSION 3 by default for Linux configurations.`

Comment (JanneKiiskila) [reply cmake/edge_configure.cmake:155]: @teetak01 - yep, does not sound right at all. But, async api sounds like something else, any ideas?

Comment (teetak01) [reply cmake/edge_configure.cmake:155]: Either use the Legacy sync API v=0, or use the improved v3.

Comment (prakharbhatt007) [reply cmake/edge_configure.cmake:155]: changing into 3 produces the same result
#error "Async PAL DNS API v2 or greater is only supported on Mbed."
 #error "Async PAL DNS API v2 or greater is only supported on Mbed."

Comment (teetak01) [reply cmake/edge_configure.cmake:155]: Remove the dependencies to Uc-hub and it will work.

File: Makefile.test (hunk 1 above)
Comment (teetak01) [reply Makefile.test:38]: You should remove the --show-reachable=yes flag too.

Comment (teetak01) [reply Makefile.test:38]: what is the reason to force the exit code to be 255 ?

File: cmake/edge_configure.cmake (hunk 4 above)
Comment (prakharbhatt007) [reply cmake/edge_configure.cmake:155]: I am not sure if we can do that as of now. We will be depreciating UC-HUB progressively but not in this build.

Comment (JanneKiiskila) [reply cmake/edge_configure.cmake:155]: Which configs would use anything but FOTA_FW now?  LmP is already fully FOTA_FW.

Comment (prakharbhatt007) [reply cmake/edge_configure.cmake:155]: This flag is just the case of UC_HUB only. In FOTA_FW, we are using the default PAL_DNS_API_VERSION. So we are using v1 in the case of UC_HUB and and the default version of MCC in FOTA_FW.

File: Makefile.test (hunk 1 above)
Comment (prakharbhatt007) [reply Makefile.test]: removed show reachable flag. I don't know the reason of the exit code, hence removed that as well.

**Review by JanneKiiskila — APPROVED** [review]

**Comment by JanneKiiskila** [comment]
@prakharbhatt007  - rebase needed, branch is out-of-date.

**Pull request merged**